internal/cloudsql/instance.go

```diff
@@ -242,90 +242,36 @@ func (c ConnectionInfo) TLSConfig() *tls.Config {
 		pool.AddCert(caCert)
 	}
 
-	// If the instance metadata does not contain a domain name, use the legacy
-	// validation checking the CN field for the instance connection name.
-	if c.DNSName == "" {
-		return &tls.Config{
-			ServerName:   c.ConnectionName.String(),
-			Certificates: []tls.Certificate{c.ClientCertificate},
-			RootCAs:      pool,
-			// We need to set InsecureSkipVerify to true due to
-			// https://github.com/GoogleCloudPlatform/cloudsql-proxy/issues/194
-			// https://tip.golang.org/doc/go1.11#crypto/x509
-			//
-			// Since we have a secure channel to the Cloud SQL API which we use to
-			// retrieve the certificates, we instead need to implement our own
-			// VerifyPeerCertificate function that will verify that the certificate
-			// is OK.
-			InsecureSkipVerify:    true,
-			VerifyPeerCertificate: verifyPeerCertificateFunc(c.ConnectionName, pool),
-			MinVersion:            tls.VersionTLS13,
-		}
-	}
-
-	// If the connector was configured with a domain name, use that domain name
-	// to validate the certificate. Otherwise, use the DNS name from the
-	// instance metadata retrieved from the ConnectSettings API endpoint.
-	serverName := c.ConnectionName.DomainName()
-	if serverName == "" {
+	var serverName string
+	if c.ConnectionName.HasDomainName() {
+		// If the connector was configured with a DNS name, use the DNS name from
+		// the configuration to validate the server certificate.
+		serverName = c.ConnectionName.DomainName()
+	} else {
+		// If the connector was configured with an Instance Connection Name,
+		// use the DNS name from the instance metadata.
 		serverName = c.DNSName
 	}
 
-	// By default, use Standard TLS hostname verification name to
-	// verify the server identity.
 	return &tls.Config{
 		ServerName:   serverName,
 		Certificates: []tls.Certificate{c.ClientCertificate},
 		RootCAs:      pool,
-		MinVersion:   tls.VersionTLS13,
+		// We need to set InsecureSkipVerify to true due to
```

Comment:
I think we should update this comment.

Historically, Cloud SQL server certificates had `my-instance:my-project` in the common name, which broke TLS validation and required `InsecureSkipVerify` being set to true. There are still instances in the Cloud SQL fleet with those certificates. So this comment is still correct for those instances. It's worth explaining why we need it for all instances now (plus any future work to improve the situation).